commit aa18367c1169700f610565eba8acf3e08429fcf5
Author: Wilco Dijkstra <wilco.dijkstra@arm.com>
Date:   Thu May 29 15:08:15 2025 +0000

    AArch64: Improve enabling of SVE for libmvec
    
    When using a -mcpu option in CFLAGS, GCC can report errors when building libmvec.
    Fix this by overriding both -mcpu and -march with a generic variant with SVE added.
    Also use a tune for a modern SVE core.
    
    Reviewed-by: Yury Khrustalev <yury.khrustalev@arm.com>

diff --git a/sysdeps/aarch64/fpu/Makefile b/sysdeps/aarch64/fpu/Makefile
index be8541f6496d6688..aa547b21df5f41d9 100644
--- a/sysdeps/aarch64/fpu/Makefile
+++ b/sysdeps/aarch64/fpu/Makefile
@@ -49,8 +49,11 @@ libmvec-support = $(addsuffix f_advsimd,$(float-advsimd-funcs)) \
                   v_powf_data
 endif
 
-sve-cflags = -march=armv8-a+sve
+# Enable SVE for building libmvec.  Since CFLAGS may contain a -mcpu or -march,
+# add a generic -mcpu and -march with SVE enabled.  Also use a tune for a modern
+# SVE core.
 
+sve-cflags = -mcpu=generic+sve -march=armv8-a+sve -mtune=neoverse-v2
 
 ifeq ($(build-mathvec),yes)
 bench-libmvec = $(addprefix float-advsimd-,$(float-advsimd-funcs)) \
